noxfiles/ci_nox.py

```diff
@@ -0,0 +1,205 @@
+        run_command = black_command
+    else:
+        run_command = (*black_command, "--check")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def isort(session: nox.Session) -> None:
+    """Run the 'isort' import linter."""
+    isort_command = ("isort", "src", "tests", "noxfiles")
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "isort")
+    elif session.posargs == ["fix"]:
+        run_command = isort_command
+    else:
+        run_command = (*isort_command, "--check-only")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def mypy(session: nox.Session) -> None:
+    """Run the 'mypy' static type checker."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "mypy")
+    else:
+        run_command = ("mypy",)
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def pylint(session: nox.Session) -> None:
+    """Run the 'pylint' code linter."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "pylint")
+    else:
+        run_command = ("pylint", "src", "noxfiles")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def xenon(session: nox.Session) -> None:
+    """Run the 'pylint' code linter."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "xenon")
+    else:
+        run_command = (
+            "xenon",
+            "src",
+            "--max-absolute B",
+            "--max-modules B",
+            "--max-average A",
+            "--ignore 'data, tests, docs'",
+            "--exclude src/fidesctl/_version.py",
+        )
+    session.run(*run_command, external=True)
+
+
+# Fidesctl Checks
+@nox.session()
+def check_install(session: nox.Session) -> None:
+    """Check that fidesctl is installed."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "check_install")
+    else:
+        run_command = ("fidesctl", *(WITH_TEST_CONFIG), "--version")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def fidesctl(session: nox.Session) -> None:
+    """Run a fidesctl evaluation."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "fidesctl")
+    else:
+        run_command = ("fidesctl", "--local", *(WITH_TEST_CONFIG), "evaluate")
+    session.run(*run_command, external=True)
+
+
+@nox.session()
+def fidesctl_db_scan(session: nox.Session) -> None:
+    """Scan the fidesctl application database to check for dataset discrepancies."""
+    session.notify("teardown")
+    session.run(*START_APP, external=True)
+    run_command = (
+        *RUN,
+        "fidesctl",
+        *(WITH_TEST_CONFIG),
+        "scan",
+        "dataset",
+        "db",
+        "postgresql+psycopg2://postgres:fidesctl@fidesctl-db:5432/fidesctl_test",
+    )
+    session.run(*run_command, external=True)
+
+
+# Pytest
+@nox.session()
+@nox.parametrize(
```

Comment:
These parameters work, but to run them you run as:

```console
nox -s "pytest(integration)"
```

rather than:

```console
nox -s pytest -- integration
```

It works so no issue, but since you mentioned wanting to make it work like the latter I just wanted to make sure this is what you want.

Reply:
correct, I changed it later and didn't mention why, my bad! I did it so that when nox lists the available sessions it is more clear to the user what is available for pytest, i.e.

```
- pytest(unit) -> Runs tests.
- pytest(integration) -> Runs tests.
- pytest(not-external) -> Runs tests.
```

instead of

```
- pytest -> Runs tests.
```